File: 파이썬/가장 먼 노드.py
from collections import deque


def solution(n, edge):
    answer = 0
    nodes = {}
    for v1, v2 in edge:
        nodes[v1] = nodes.get(v1, []) + [v2]
        nodes[v2] = nodes.get(v2, []) + [v1]
    que = deque([(1, 0)])
    cand = []
    visited = set([1])
    while que:
        cur_node, dist = que.popleft()
        for nxt_node in nodes[cur_node]:
            if not nxt_node in visited:
                que.append((nxt_node, dist+1))
                visited.add(nxt_node)
        cand.append(dist)
    max_dist = max(cand)
    return cand.count(max_dist)

File: 자바스크립트/가장 먼 노드.js
function solution(n, edge) {
    let visited = new Set([1])
    let nodeTable = {};
    let dists = []
    let que = [[1, 1]]
    
    for (const [a, b] of edge) {
        nodeTable[a] = (nodeTable[a] || []).concat([b])
        nodeTable[b] = (nodeTable[b] || []).concat([a])
    }

    while (que.length > 0) {
        const [node, dist] = que.shift();
        const nexts = nodeTable[node]
        dists.push(dist)

        for (const nxtNode of nexts) {
            if (!visited.has(nxtNode)) {
                visited.add(nxtNode)
                que.push([nxtNode, dist+1])
            }
        }
    }
    
    let max = Math.max(...dists)
    
    return dists.filter(x => x === max).length
}